cmake/kokkos_tribits.cmake

```diff
@@ -288,6 +306,43 @@ function(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
     target_compile_options(
       ${LIBRARY_NAME} PUBLIC $<$<COMPILE_LANGUAGE:${KOKKOS_COMPILE_LANGUAGE}>:${NODEDUP_XCOMPILER_OPTIONS}>
     )
+    list(APPEND ALL_KOKKOS_COMPILER_FLAGS ${NODEDUP_XCOMPILER_OPTIONS})
+  endif()
+
+  message(STATUS "Checking if compiler can consume CMAKE_${KOKKOS_COMPILE_LANGUAGE}_FLAGS and Kokkos compile flags")
+  list(APPEND ALL_COMPILER_FLAGS ${CMAKE_${KOKKOS_COMPILE_LANGUAGE}_FLAGS})
```

Comment:
```suggestion
```

Reply:
Even if `CMAKE_<LANGUAGE>_FLAGS` are tested individually, we probably want to check if the combination of the flags we set and the user given ones create a set that the compiler can consume.

Reply:
Yes, of course. That's what's happening already if you only test with Kokkos compile flags explicitly since `MAKE_<LANGUAGE>_FLAGS` is used implicitly.